test/unit/common/msRestRequestTest.js

```diff
@@ -0,0 +1,102 @@
+var should = require('should');
+var sinon = require('sinon');
+
+var msRestRequest;
+var Token = require('../../../lib/common/token');
+
+describe('msRestRequest', function() {
+  before(function(){
+    sinon.stub(Token, 'getToken').yields(null, 'fake-token');
+  });
+  
+  after(function(){
+    Token.getToken.restore();
+  });
+  
+  describe('when the token is not expired and statusCode is not in AZURE_RETRY_ERROR_CODES', function() {
+    
+    before(function() {
+      require('request');
+      require.cache[require.resolve('request')].exports = function(_, callback) {
```

Comment:
@niroyb 
But it doesn't support to detect how many times the method is called so far. We can use it after it is mature enough.